org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/model/BinaryFileAccess.java

```diff
@@ -0,0 +1,52 @@
+/**
+ * Copyright (c) 2017, 2020 itemis AG (http://www.itemis.eu) and others.
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ * 
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.xtext.xtext.generator.model;
+
+import java.io.ByteArrayInputStream;
+import java.util.Arrays;
+
+import org.eclipse.xtext.generator.IFileSystemAccess2;
+
+/**
+ * A utility class for generating binary files.
+ */
+public class BinaryFileAccess {
+	private String path;
+
+	protected byte[] internalContents;
+
+	public void setContent(final byte[] content) {
+		this.internalContents = content;
+	}
+
+	public byte[] getContent() {
+		return Arrays.copyOf(internalContents, internalContents.length);
```

Comment:
Copy should probably happen on setContent, too.
Can be simplified to return internalContents.clone

Reply:
?

Reply:
No worries :) It would only spare us a few characters in the source code but not improve much.